core/msg.c

```diff
@@ -370,6 +370,25 @@ static int _msg_receive(msg_t *m, int block)
     DEBUG("This should have never been reached!\n");
 }
 
+int msg_avail(void)
+{
+    unsigned state = disableIRQ();
+    DEBUG("msg_available: %" PRIkernel_pid ": msg_available.\n",
+          sched_active_thread->pid);
+
+    tcb_t *me = (tcb_t*) sched_threads[sched_active_pid];
```

Comment:
In here you can use `sched_active_thread`. For the other functions we need to look if they [are/may be] used in interrupt handlers. I am quite sure they must not be called by an interrupt handler, because it would simply make no sense if I am not mistaken.